### PR TITLE
STCOR-689 STCOR-690 Support switch consortium active affiliation

### PR DESCRIPTION
**Pull request by NikitaSedyx**
## Purpose
https://issues.folio.org/browse/STCOR-689
https://issues.folio.org/browse/STCOR-690

As a central tenant of a consortium, it should be able to switch the current context to manage institutional tenants. Header `X-Okapi-Tenant` acts as a pointer to the current context when sending requests to the server.

---

If a user login into the consortium central tenant, he expects to see its name in the top-right corner of the screen (in the profile dropdown trigger, above a current SP name).

## Approach
- Validate user based on current tenant value stored in a session (if absent - use tenant from stripes configs);
- Expose `updateTenant` service, which will update the current tenant, fetch resources (perms, SPs and etc.) and update session;
- Allow `<HandlerManager>` to invoke handlers for other module types (`app`, `settings`);
- Render active tenant (affiliation) name in the profile dropdown trigger;
- Implement unit test;

<details open><summary>Preview</summary>
<p>

firefox_Wtyh0AQ9Zx

</p>
</details>

### CODE REVIEW

**Review by usavkov-epam — APPROVED** [review]

**Comment by NikitaSedyx** [comment]
hi @mkuklis  @zburke please review one more implementation for consortia stuff (follows recommendation to keep it agnostic to compare with https://github.com/folio-org/stripes-core/pull/1307)

File: src/components/HandlerManager/HandlerManager.js [hunk 1]
```diff
@@ -20,7 +20,15 @@ class HandlerManager extends React.Component {
   constructor(props) {
     super(props);
     const { event, stripes, modules, data } = props;
-    this.components = getEventHandlers(event, stripes, modules.handler, data);
+    const uniqueModules = Object.values(modules)
```
Comment (mkuklis) [inline src/components/HandlerManager/HandlerManager.js:23]: @NikitaSedyx I think the original idea for this was to only support the `handler` modules. Could you explain why was this changed here?

Comment (NikitaSedyx) [reply src/components/HandlerManager/HandlerManager.js:23]: because non-handler modules can't handle events like `LOGIN`, according to https://github.com/folio-org/stripes/blob/master/doc/dev-guide.md#handlers-and-events 
```
Stripes modules of any type can provide a handler; the handler module type is provided for the case where the handler is the only thing the module provides, i.e. there is no main-screen app and no settings.
```
and it's strange that we provide a handler in app or settings module, but it can't handle events

Comment (NikitaSedyx) [reply src/components/HandlerManager/HandlerManager.js:23]: as an alternative, we can add `handler` type to our module, but it will contradict with description of handler module, additionally it will require to refactor ProfileDropdown logic to handle duplicates (actually the same logic that is applied here)

IMO following current documentation,  HandlerManager should support handler of all module types, not just handlers

Comment (zburke) [reply src/components/HandlerManager/HandlerManager.js:23]: I agree with @mkuklis here. I don't think this needs to change; you just need to add the `handler` to the `stripes.actsAs` array in the `package.json` of the ui-$whatever repo. Have you tried that?

Comment (NikitaSedyx) [reply src/components/HandlerManager/HandlerManager.js:23]: yes, in this case we have duplicates in ProfileDropdown actions because it lists action for each value in `actsAs` - as I mentioned above it's not a problem as we just need to add module uniqueness 

but here I have another point, for action menu we can add config to any module, define handler to return component and it will work without `stripes.actsAs: handler`, at the same time to handle events via `HandlerManager` we need to define module as a handler, it's usage inconsistency

Comment (NikitaSedyx) [reply src/components/HandlerManager/HandlerManager.js:23]: so we have to use cases to use handlers one doesn't require module as a handler, another one requires

Comment (NikitaSedyx) [reply src/components/HandlerManager/HandlerManager.js]: @zburke @mkuklis reverted changes for HandlerManager, probably handler docs section should be updated, at least 2 devs were confused by it :)

Comment (mkuklis) [reply src/components/HandlerManager/HandlerManager.js]: Thank you @NikitaSedyx. I agree with you that the docs should include the information about how to turn the existing ui-module into a `handler` and that the `HandlerManager` is intended to be used only by `hander` modules. Do you think that would be sufficient?

Comment (NikitaSedyx) [reply src/components/HandlerManager/HandlerManager.js]: I believe it should be enough, I was a bit confused by 2 points
1. any module can provide handler (but in fact ignored by `HandlerManager`)
2. handler type is used for case when module provides only handler

looks like what you mentioned covers them

Comment (mkuklis) [reply src/components/HandlerManager/HandlerManager.js]: Thank you @NikitaSedyx

**Review by mkuklis — APPROVED** [review]

File: src/components/MainNav/ProfileDropdown/ProfileDropdown.js [hunk 2]
```diff
@@ -228,27 +242,40 @@ class ProfileDropdown extends Component {
...
     );
   }
 
+  renderProfileTriggerLabel = ({ open }) => {
+    const { okapi } = this.props.stripes;
+    const userData = this.getUserData();
+    const servicePointName = userData?.curServicePoint?.name;
+    const tenantName = userData?.tenants?.find(({ id }) => id === okapi.tenant)?.name;
+
+    const withLabel = Boolean(servicePointName || tenantName);
```
Comment (zburke) [inline src/components/MainNav/ProfileDropdown/ProfileDropdown.js:264]: A better convention than `with...` is `is...` or `has...`. We use `withFoo` as a naming convention for HOCs that provide a `Foo` object to the component they wrap.

**Review by zburke — APPROVED** [review]

File: src/components/MainNav/ProfileDropdown/ProfileDropdown.js (hunk 2 above)
Comment (NikitaSedyx) [reply src/components/MainNav/ProfileDropdown/ProfileDropdown.js]: agree, missed this migrating from https://github.com/folio-org/stripes-core/pull/1307 PR
